platform/api.search/src/org/netbeans/modules/search/Constants.java

```diff
@@ -29,11 +29,11 @@
     /**
      * maximum number of found objects
      */
-    public static final int COUNT_LIMIT = 500;
+    public static final int COUNT_LIMIT = Integer.getInteger("netbeans.search.count.limit", 500);
```

Comment:
This shall be documented in `arch.xml` as an `<api group="systemproperty" ...>some description</api>`. See [howto](http://wiki.netbeans.org/APIDevelopment#Writing_the_documentation) for more details about `arch.xml`. 

Also one shall version the change. E.g. increase the module version and leave a note in `apichanges.xml` file.